File: leetcode/24. Swap Nodes in Pairs.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if (not head or not head.next):
            return head
        n1 = head
        n2 = head.next
        n1.next = self.swapPairs(n2.next)
        n2.next = n1
        return n2

        

File: leetcode/24. Swap Nodes in Pairs.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class ListNode {
  int val;
  ListNode? next;
  ListNode([this.val = 0, this.next]);
}

class Solution {
  ListNode? swapPairs(ListNode? head) {
    if (head == null || head.next == null) return head;
    ListNode? n1 = head, n2 = head.next;
    n1.next = swapPairs(n2!.next);
    n2.next = n1;
    return n2;
  }
}
